public/app/features/variables/utils.ts

```diff
@@ -137,7 +137,7 @@ export function getTemplatedRegex(variable: QueryVariableModel, templateSrv = ge
 
 export function getLegacyQueryOptions(variable: QueryVariableModel, searchFilter?: string, timeSrv = getTimeSrv()) {
   const queryOptions: any = { range: undefined, variable, searchFilter };
-  if (variable.refresh === VariableRefresh.onTimeRangeChanged) {
+  if (variable.refresh === VariableRefresh.onTimeRangeChanged || variable.refresh === VariableRefresh.onDashboardLoad) {
```

Comment:
I think tests can be added to https://github.com/grafana/grafana/blob/master/public/app/features/variables/query/queryRunners.test.ts

Reply:
thanks! 🙏

Reply:
added a test case in `queryRunners.test.ts` but I'm still not sure how to properly test that a variable with `refresh = onDashboardLoad` won't get refreshed when time range changes

Reply:
For this case I think you need to look into https://github.com/grafana/grafana/blob/master/public/app/features/variables/state/onTimeRangeUpdated.test.ts